pkg/querier/http.go

```diff
@@ -189,9 +187,16 @@ func (q *QuerierAPI) LogQueryHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := marshal_legacy.WriteQueryResponseJSON(result, w); err != nil {
+	if r.Header.Get("Accept") == queryrange.ProtobufType {
+		w.Header().Add("Content-Type", queryrange.ProtobufType)
+		err = queryrange.WriteQueryResponseProtobuf(&params, result, w)
+	} else {
+		w.Header().Add("Content-Type", queryrange.JSONType)
+		err = marshal_legacy.WriteQueryResponseJSON(result, w)
+	}
```

Comment:
I kept this instead of `queryrange.WriteResponse` because it's not easy to see when to use `marshal_legacy`.